src/app/repositories/firebase/dataset_firebase_repository.py

```diff
@@ -105,3 +105,49 @@ def get_dataset_metadata_collection(
             .where("period_id", "==", period_id)
             .stream()
         )
+
+    def delete_previous_dataset_versions(
+        self, survey_id: str, latest_version: int
+    ) -> None:
+        """
+        Queries firestore for older versions of a dataset associated with a survey id,
+        iterates through them and deletes them and their subcollections recursively. The
+        recursion is needed because you cannot delete subcollections of a document in firestore
+        just by deleting the document, it does not cascade.
+
+        Parameters:
+        survey_id (str): survey id of the dataset.
+        latest_version (int): latest version of the dataset.
+        """
+
+        previous_dataset_versions = self.datasets_collection.where(
```

Comment:
Just a minor thing. Perhaps change the var name previous_dataset_versions into previous_version_datasets or previous_version_datasets_collection as the name confuse me that it is a collection of version integers